### PR TITLE
Cities are shown in cities tab; Nested routes cities/:cityId implemented + loading states added

### PR DESCRIPTION
**Pull request by ziviledauderiene**
screenshot034

- Current weather data of 4/8 cities are shown in the cities tab.
- "Load more/show less" button to show more/less cities.
- Clicking on "show full forecast" navigates to all forecast info of the selected city.
- Nested routes "/cities/:cityId" implemented. The correct weather info is shown using ":cityId" param.
- Location-context deleted.
- Loading states with spinning sun added.

### CODE REVIEW

File: src/components/App/App.tsx [hunk 1]
```diff
@@ -10,16 +10,19 @@ import Header from '../Header/Header';
...
   useEffect(() => {
     if (!document.cookie) {
-      axios.get(ENDPOINTS.auth, {
-        withCredentials: true
-      });
+      axios
+        .get(ENDPOINTS.auth, {
+          withCredentials: true
+        })
+        .then(() => location.reload());
```
Comment (JuliaShavyrova) [inline src/components/App/App.tsx]: why do you need to reload page?

File: src/helpers/cities.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+const cities: { name: string; country: string; id: string }[] = [
```
Comment (JuliaShavyrova) [inline src/helpers/cities.ts]: seems like you have the same interface here and inside CityProps. please make separate interface for city

File: src/helpers/date.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+export const getWeekDay = (day: number): string => {
```
Comment (JuliaShavyrova) [inline src/helpers/date.ts]: please use native js methods to get week days

File: src/components/UI/Title/Title.tsx [hunk 4]
```diff
@@ -1,14 +1,22 @@
...
 }
 
-const Title: React.FunctionComponent<TitleProps> = ({ title }) => {
+const Title: React.FunctionComponent<TitleProps> = ({ title, fontSize }) => {
   const { theme }: ThemeContextConfig = useContext(ThemeContext);
-  return <h2 className={`${styles.title} ${theme === Theme.DARK && styles[theme]}`}>{title}</h2>;
+  return (
+    <h2
+      className={`${styles.title} ${theme === Theme.DARK && styles[theme]}`}
+      style={{ fontSize: fontSize }}
```
Comment (JuliaShavyrova) [inline src/components/UI/Title/Title.tsx]:
```suggestion
      style={{ fontSize }}
```

File: src/components/Cities/Cities.tsx [hunk 5]
```diff
@@ -1,7 +1,41 @@
...
 const Cities: React.FunctionComponent = () => {
-  return <div>Cities</div>;
+  const { cityId } = useParams<string>();
+  const [showMore, setShowMore] = useState<boolean>(false);
+  const clickHandler = () => setShowMore(showMore => !showMore);
+  const citiesToShow = cities.length / 2;
+
+  return (
+    <>
+      {!cityId ? (
```
Comment (JuliaShavyrova) [inline src/components/Cities/Cities.tsx]: why not?
{!cityId && <>....</>}

File: src/helpers/date.ts (hunk 3 above)
Comment (ziviledauderiene) [reply src/helpers/date.ts]: ok, changed 
`${new Date(data.current.time).toLocaleDateString('en-US', { weekday: 'long' })}`}`

**Review by JuliaShavyrova — APPROVED** [review]

**Pull request merged**